tony-portal/app/history/HistoryFileMover.java

```diff
@@ -0,0 +1,87 @@
+package history;
+
+import com.linkedin.tony.Constants;
+import com.linkedin.tony.TonyConfigurationKeys;
+import com.linkedin.tony.util.ParserUtils;
+import com.linkedin.tony.util.Utils;
+import com.typesafe.config.Config;
+import hadoop.Requirements;
+import java.io.IOException;
+import java.util.Date;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import play.Logger;
+import utils.ConfigUtils;
+
+
+@Singleton
+public class HistoryFileMover {
+  private static final Logger.ALogger LOG = Logger.of(HistoryFileMover.class);
+
+  private final FileSystem fs;
+  private final Path intermediateDir;
+  private final Path finishedDir;
+
+  @Inject
+  public HistoryFileMover(Config appConf, Requirements requirements) {
+    fs = requirements.getFileSystem();
+    intermediateDir = requirements.getIntermDir();
+    finishedDir = requirements.getFinishedDir();
+
+    ScheduledExecutorService scheduledThreadPool = Executors.newScheduledThreadPool(1);
+    long moverIntervalMs = ConfigUtils.fetchIntConfigIfExists(appConf,
```

Comment:
fetchLongConfigIfExists?

Reply:
The method actually returns an `int` and then I just rely on Java's auto conversion to `long`.